src/main/java/fr/xephi/authme/listener/AuthMePlayerListener.java

```diff
@@ -428,13 +428,21 @@ public void onPlayerLogin(PlayerLoginEvent event) {
             }
         }
 
-        if (Settings.isKickNonRegisteredEnabled) {
+        if (Settings.isKickNonRegisteredEnabled && !Settings.antiBotInAction){
             if (!plugin.database.isAuthAvailable(name)) {
                 event.setKickMessage(m.send("reg_only")[0]);
                 event.setResult(PlayerLoginEvent.Result.KICK_OTHER);
                 return;
             }
         }
+        
+        if (Settings.antiBotInAction){
+            if (!plugin.database.isAuthAvailable(name)) {
+                event.setKickMessage(m.send("AntiBot service in action! Non registered players can't connect until the bot attack stops!")[0]); //Need to add string to messages
```

Comment:
you will get a null pointer by calling this ^^", cause that string doesn't exist